src/box/lua/tuple.lua

```diff
@@ -107,24 +108,24 @@ local encode_fix = msgpackffi.internal.encode_fix
 local encode_array = msgpackffi.internal.encode_array
 local encode_r = msgpackffi.internal.encode_r
 
-local tuple_encode = function(tmpbuf, obj)
+local tuple_encode = function(tmpbuf, obj, level)
     if obj == nil then
         encode_fix(tmpbuf, 0x90, 0)  -- empty array
     elseif is_tuple(obj) then
-        encode_r(tmpbuf, obj, 1)
+        encode_r(tmpbuf, obj, 1, advance_trace(level))
```

Comment:
Used `level and level + 1` instead of `advance_trace(level)` as discussed f2f. Inside function may need to pass `level` or `level + 1` due to tail call optimization.